File: server/Models/Keep.cs
using System.ComponentModel.DataAnnotations;
using keepr.interfaces;

namespace keepr.Models;

// RETURN DATA 
public class Keep : IRepoItem<int>
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string Name { get; set; }
    [MaxLength(1000)]
    public string Description { get; set; }
    [MaxLength(1000)]
    public string Img { get; set; }
    public int Views { get; set; }

    // RELATIONSHIP properties
    public string CreatorId { get; set; }
    public int Kept { get; set; }
    public Profile Creator { get; set; }
}

public class KeepCreationDTO
{
    public string Name { get; set; }
    [MaxLength(1000)]
    public string Description { get; set; }
    [MaxLength(1000)]
    public string Img { get; set; }
}

File: client/src/models/Keep.js
import { Profile } from "./Profile.js"

export class Keep{
    constructor(data){
        this.id = data.id
        this.kept = data.kept
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.creatorId = data.creatorId
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
        this.creator = data.creator ? new Profile(data.creator) : null
    }
}
